### PR TITLE
[feat] add Minimax H3 sft pipeline

### PR DESCRIPTION
**Pull request by Davids048**
## Purpose

Add joint video-and-audio supervised fine-tuning for MiniMax H3 through the modular `fastvideo.train` framework. The implementation builds on the MiniMax H3 inference support from #1674.

The training path consumes precomputed text embeddings, video latents, and stereo-audio latents. It applies independent video and audio noise schedules, and then trains both H3 output branches with flow-matching mean-squared error. The PR also provides a reproducible single-sample Crush-Smol overfit recipe with online Weights & Biases validation.

## Changes

- Add the modular `MiniMaxH3Model` training adapter, including H3 packed-token construction, video/audio noise preparation, sequence-parallel metadata, activation checkpointing, and model-specific backward context restoration.
- Extend `FineTuneMethod` and `TrainingBatch` with a joint video/audio prediction contract. The total H3 loss is the sum of the video and audio flow-matching losses. Video-only fine-tuning uses its single-tensor loss contract.
- Add the text-to-video-and-audio Parquet schema and a fixed preprocessing workflow for one Crush-Smol audio-video-caption record.
- Add H3 training dtype handling for FastVideo's Fully Sharded Data Parallel loader and device-safe materialization of non-persistent rotary-position buffers.
- Extend modular validation with validation-before-training, exact step scheduling, synchronized audio-video MP4 encoding, tracker artifact counts, and temporary training-state offload during inference.
- Add the 400-step H3 overfit configuration for an 8-by-8 hybrid sharded data parallel mesh on 64 H200 GPUs. The configuration validates at step 0 and every 20 optimizer steps with online Weights & Biases logging.
- Make `examples/train/run_slurm.sh` preserve inherited W&B authentication, accept optional node constraints and time limits, assign each node its Slurm process rank, and shell-escape generated training arguments.
- Add focused tests for preprocessing, H3 batch construction, joint loss, attention backend selection, validation scheduling, audio-video media encoding, distributed launch arguments, and configuration resolution.

## Test Plan

```bash
/mnt/weka/shrd/wm/junda/fv-hub/.venv/bin/python -m pytest -q fastvideo/tests/train/callbacks/test_validation.py fastvideo/tests/train/methods/test_minimax_h3_finetune.py fastvideo/tests/train/utils/test_moduleloader_attention_backend.py fastvideo/tests/train/utils/test_validation_media.py fastvideo/tests/workflow/test_minimax_h3_overfit_preprocess.py

git diff --check main...HEAD
bash -n examples/train/run_slurm.sh

git diff --name-only -z main...HEAD | xargs -0 /mnt/weka/shrd/wm/junda/fv-hub/.venv/bin/pre-commit run --files
```

The experiment configuration can be submitted with:

```bash
SBATCH_CONSTRAINT=nvidia_h200 SBATCH_TIMELIMIT=72:00:00 WANDB_MODE=online bash examples/train/run_slurm.sh examples/train/configs/overfit_minimax_h3_t2va.yaml 8
```

## Test Results

<details>
<summary>Focused test and static-check output</summary>

```text
  71 passed, 16 warnings in 1.37s

  yapf.....................................................................Passed
  ruff (legacy alias)......................................................Passed
  codespell................................................................Passed
  PyMarkdown...........................................(no files to check)Skipped
  Lint GitHub Actions workflow files...................(no files to check)Skipped
  mypy.....................................................................Passed
  Check for spaces in all filenames........................................Passed
  Suggestion...............................................................Passed

  git diff --check origin/main...HEAD: passed
  bash -n examples/train/run_slurm.sh: passed
  bash -n examples/train/launch_minimax_h3_t2va_crush_smol_validation.sh: passed
  bash -n examples/train/setup_minimax_h3_t2va_crush_smol_single_sample.sh: passed

```

The warnings are dependency deprecations from SWIG bindings and `torch.jit.script_method`.

The failing test expects a dedicated H3 launcher that is absent from `HEAD`.

</details>

### MiniMax H3 single-sample overfit

- Slurm job: `2026587`
- Output directory: `/mnt/weka/shrd/wm/junda/fv-hub/fastvideo-add-h3-sft/runs/minimax_h3_t2va_crush_smol_single_sample_overfit`
- W&B run: [js202/fastvideo_minimax_h3/59543p6e](https://wandb.ai/js202/fastvideo_minimax_h3/runs/59543p6e)
- Input video: `data/crush-smol/videos/1gGQy4nxyUo-Scene-016.mp4`
- Input caption: "A watermelon wearing a helmet is crushed by a hydraulic press, causing it to flatten and burst open."
- Training targets: 124 video frames at 768 x 1344 and 24 FPS, plus the synchronized stereo soundtrack resampled to 32 kHz.
- Allocation: 8 nodes, 64 NVIDIA H200 GPUs, 1,024 CPU cores, and 11,520 GiB of host memory. The 8-by-8 mesh provided eight data-parallel replicas and an effective global batch size of 8.
- Compute usage: 3:05:19 elapsed, 197.67 allocated H200 GPU-hours, and 24.71 allocated node-hours.
- Result: Validation in the linked Weights & Biases run showed that MiniMax H3 overfits the single Crush-Smol sample.

## Checklist

- [ ] I ran `pre-commit run --all-files` and fixed all issues
- [x] I added or updated tests for my changes
- [x] I updated documentation if needed
- [x] I considered GPU memory impact of my changes

**For model/pipeline changes, also check:**

- [ ] I verified SSIM regression tests pass
- [x] I updated the support matrix if adding a new model (N/A: MiniMax H3 inference support was added in #1674).

### CODE REVIEW

**Review by SolitaryThinker — CHANGES_REQUESTED** [review]
Two scripts this PR depends on aren't in the diff: `examples/train/launch_minimax_h3_t2va_crush_smol_validation.sh` and `examples/train/setup_minimax_h3_t2va_crush_smol_single_sample.sh`. `test_h3_slurm_scripts_preserve_inherited_wandb_key` fails on head because of it (reproduced: 1 failed / 70 passed), and that file is in the train-framework lane selection, so the lane will be red. The overfit config's header comment also points at the missing launcher. Looks like a forgotten `git add` — push the scripts or drop those assertions + fix the comment.

Asks:
- `fastvideo/train/utils/activation_checkpoint.py` is a formatting-only copy of `fastvideo/training/activation_checkpoint.py`; wan/ltx2/kandinsky5 still import the legacy one. Import the existing module like the siblings, or move it for everyone — two copies will drift.
- ValidationCallback now forces `dit_cpu_offload=False` for every recipe (base set True via `make_inference_args`). Right call for the FSDP-shared transformer, but it's a silent memory-behavior change for all existing validation — say so in the PR description.
- `bash -n a b c` only syntax-checks the first file — the launcher/setup scripts are never actually checked in that test.
- Audio encode failure now aborts the validation event (and training) instead of skipping the artifact — intentional? A transient ffmpeg failure killing a 64-GPU run seems harsh.
- FYI: WANDB_API_KEY must now be *exported* in the submitting shell (that's the point — it no longer leaks into xtrace logs), header docs still read as before. And #1689 also edits `fastvideo/configs/models/dits/minimax_h3.py`; whoever lands second rebases.

Verified locally on CPU: the three fixes (RoPE inv_freq device-follow, buffer materialization on device change, stale image_path leak across validation records) all fail on main and pass on this head; unit/contract/api selections green apart from the launcher test.

**Comment by Davids048** [comment]
> fastvideo/train/utils/activation_checkpoint.py is a formatting-only copy of fastvideo/training/activation_checkpoint.py; wan/ltx2/kandinsky5 still import the legacy one. Import the existing module like the siblings, or move it for everyone — two copies will drift.

We should slowly move away from the old `training` framework and towards the new yml config based one.
